src/QMCDrivers/tests/test_EngineHandle.cpp

```diff
@@ -46,7 +46,8 @@ TEST_CASE("EngineHandle construction", "[drivers]")
 
 
   int fake_num = 5;
-  handle->prepareSampling(fake_num);
+  int fake_sample_num = 100;
+  handle->prepareSampling(fake_num,fake_sample_num);
```

Comment:
this needs to be tested in a threaded context.

Reply:
in fact all of the engine methods need to be or it needs to be.  Or they need to be called only from one thread.

Reply:
It prepares storage for each sample.
This is called outside the thread context and thread agnostic.